### PR TITLE
Use Hugo USWDS theme as a Hugo module

### LINKED ISSUE #349: Transition USWDS Hugo theme to use Hugo Modules
### User Story

As an Metaschema developer, I would like the Hugo site to be transitioned to Hugo modules in order to simplify CI/CD (and remove a submodule from the repository)

### Goals

- [ ] All local CI/CD and local test infrastructure should be transitioned to rely on Hugo submodules
- [ ] The submodule should be removed
- [ ] A Wiki page or readme should be created explaining how to update the hugo module & stub a replacement for local theme development

### Dependencies

N/A

### Acceptance Criteria

- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

### PR DESCRIPTION
**Pull request by david-waltermire-nist**
# Committer Notes

This PR updates the Hugo configuration and supporting files to use the USWDS theme as a Hugo module.

This work was moved out of PR #393.

Some general cleanup of the repository is also included.

Resolves #349.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you included examples of how to use your new feature(s)?
- [x] Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.

### CODE REVIEW

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ on:
         required: true
         default: 'published'
         type: string
-      create_issue:
+      linkcheck_create_issue:
         description: 'create new GitHub issue if broken links found'
         required: false
         default: true
```
Comment (aj-stein-nist) [inline .github/workflows/link-check.yml:32]: Non-blocking: not immediately applicable to this PR and not not this block of code, this block is just the most relevant. Nikita figured out an ingenious way to keep posting to the same issue instead of spamming the repo with a new issue per failure, you want to check this out within this PR or a follow-up one. It made the team less upset in the OSCAL work. 😄 

https://github.com/usnistgov/OSCAL/pull/1916

File: .github/workflows/config/.markdown-link-check/config.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  ],
+  "replacementPatterns": [
+        {
+            "pattern": "^../../issues",
+            "replacement": "https://github.com/usnistgov/metaschema/issues"
+        },
+        {
+            "pattern": "^../../projects",
+            "replacement": "https://github.com/usnistgov/metaschema/projects"
+        },
```
Comment (aj-stein-nist) [inline .github/workflows/config/.markdown-link-check/config.json:24]: Non-blocking: is Metaschema not using a [projects board](https://github.com/orgs/usnistgov/projects/44) that not a repo-based [projects (classic} board](https://github.com/usnistgov/metaschema/projects/1)? Is that mapping still applicable or appropriate?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
+This project uses an agile approach for development. We’re trying to focus on the core capabilities that are needed to provide the greatest amount of benefit. Because we’re working on a small set of capabilities, this allows us to make very fast progress. We’re building the features that we believe solve the biggest problems to provide the most value to the most people. We provide extension points that allow uncovered cases to be supported by others.
 
-We track our current work items using GitHub [project cards](../../projects). Our active project is typically the lowest numbered open project within the previously referenced page.
+We track our current work items using GitHub [project cards](../../projects).
```
Comment (aj-stein-nist) [inline CONTRIBUTING.md:9]: Non-blocking: re my comment about the new (not classic) [projects board](https://github.com/orgs/usnistgov/projects/44) board? This link only works because we didn't point to the specific board.

```suggestion
We track our current work items using GitHub [project cards](https://github.com/orgs/usnistgov/projects/44).
```

(Also, the board is not public, but that is orthogonal to the PR.)

File: USERS.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Users Guide
```
Comment (aj-stein-nist) [inline USERS.md:1]: Non-blocking: are we going to further explain what kind of users are the target of this project in future work?

**Review by aj-stein-nist — APPROVED** [review]
LGTM. I only have minor, non-blocking commentary.

File: USERS.md (hunk 4 above)
Comment (david-waltermire-nist) [reply USERS.md:1]: Yes. Maybe in the future?

Comment (aj-stein-nist) [reply USERS.md:1]: Non-blocking means I approve and all comments are at your discretion for now or later. I am resolving this one.

File: .github/workflows/link-check.yml (hunk 1 above)
Comment (aj-stein-nist) [reply .github/workflows/link-check.yml:32]: Discussed in chat.

File: .github/workflows/config/.markdown-link-check/config.json (hunk 2 above)
Comment (david-waltermire-nist) [reply .github/workflows/config/.markdown-link-check/config.json:24]: I'll update this.
